lib/db.js

```diff
@@ -479,19 +482,50 @@ module.exports = (
       uaDeviceType: token.uaDeviceType,
       lastAccessTime: token.lastAccessTime,
       createdAt: token.createdAt
-    }]
+    }
     let sessionTokens = []
+    let matchedTokens = []
+    const ipHash = ip && crypto.createHash('sha256').update(ip).digest('base64')
```

Comment:
If we are okay with doing an ip lookup each time it removes the need to store the hashed ip at all 👍